### PR TITLE
Replace message in 'about me' tab with info taken from DataServlet

### PR DESCRIPTION
**Pull request by andrewm898**
This code replaces the text in the 'about me' tab with text received from the server.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,6 +27,6 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Andrew!</h1>");
```
Comment (belraquib) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: Why is this changed?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -82,10 +82,42 @@ class SlideShow {
...
 document.addEventListener('DOMContentLoaded', () => {
+    getMessage();
     const container = document.getElementById('photos');
     const slideshow = new SlideShow(document.getElementById('photos'));
 });
 
+/**
+ * Fetches a message from the server to add to the page.
+ */
+function getMessage() {
```
Comment (belraquib) [inline portfolio/src/main/webapp/script.js:93]: Let's learn about async functions!

call this async function updateQuotesFromServer() 

now:
 
async function updateQuotesFromServer()  {
   const response = await fetch('/data');
   const text = await response.text();
   // your code from addQuoteToDom here.
}

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -82,10 +82,42 @@ class SlideShow {
...
+ */
+function handleResponse(response) {
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addQuoteToDom);
+}
```
Comment (mickrobk) [inline portfolio/src/main/webapp/script.js:113]: Could you pull this inline to the getMessage function with an arrow function? Rationale:
 - "handleResponse" is quite generic as a function name
 - Moving it inline would keep the logic encapsulated and close together visually, which is good for readability. There's not so much here that the function would end up too long

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -82,10 +82,42 @@ class SlideShow {
...
+
+/** Adds a quote to the DOM. */
+function addQuoteToDom(quote) {
```
Comment (mickrobk) [inline portfolio/src/main/webapp/script.js:116]: nit: Should this be "update" instead of "add"?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (andrewm898) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: This was because I was receiving the message from the DataServlet with response.text() and then writing it directly to the inner text of a div instead of creating a new header and appending it as a child. Should I do it as the latter instead?

Comment (belraquib) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: ack got it.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (belraquib) [reply portfolio/src/main/webapp/script.js:93]: I don't see a change?

File: portfolio/pom.xml [hunk 5]
```diff
@@ -23,6 +23,11 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.6</version>
+    </dependency>
```
Comment (mickrobk) [inline portfolio/pom.xml:30]: Could you pull this into it's own PR?

Comment (andrewm898) [reply portfolio/pom.xml]: xml file has been reverted to original here/changes moved to separate branch
